ddtrace/__init__.py

```diff
@@ -24,11 +29,15 @@
 ]
 
 
-@remove(removal_version="1.0.0")
+if asbool(os.getenv("DD_TRACE_RAISE_DEPRECATIONWARNING")):
+    warnings.filterwarnings(action="error", category=DDTraceDeprecationWarning)
```

Comment:
```suggestion
if asbool(os.getenv("DD_TRACE_RAISE_DEPRECATIONWARNING")):
    warnings.filterwarnings(action="error", category=DDTraceDeprecationWarning)
```

I can add as a follow-up PR, `DD_TRACE_LOG_DEPRECATIONWARNING` could be a nice to have too, `warnings.filterwarnings(action="default", category=DDTraceDeprecationWarning)`